core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -933,6 +948,22 @@ private SqlLibraryOperators() {
           ReturnTypes.BIGINT_NULLABLE, OperandTypes.TIMESTAMP,
           SqlFunctionCategory.TIMEDATE);
 
+  /** BigQuery's "DATETIME_ADD(timestamp, interval) function; Behaves similarly
+   * to BigQuery's TIMESTAMP_ADD because in Calcite, datetime is a type alias
+   * for timestamp. */
+  @LibraryOperator(libraries = {BIG_QUERY})
+  public static final SqlFunction DATETIME_ADD =
+      TIMESTAMP_ADD2.withName("DATETIME_ADD");
+
+  /** BigQuery's "DATETIME_DIFF(timestamp, timestamp, timeUnit) function; Behaves
+   * similarly to BigQuery's TIMESTAMP_DIFF because in Calcite, datetime is a type
+   * alias for timestamp. Returns the whole number of timeUnit between datetime
+   * and datetime2, with the result being negative if datetime occurs before datetime2. */
+  @LibraryOperator(libraries = {BIG_QUERY})
+  public static final SqlFunction DATETIME_DIFF =
```

Comment:
As TIMESTAMP_ADD. Also change signature to DATETIME_DIFF(timestamp, timestamp2, timeUnit)" and fix references to "datetime" and "datetime2" in the description.